lib/auth/github.go

```diff
@@ -84,6 +91,124 @@ func (a *Server) upsertGithubConnector(ctx context.Context, connector types.Gith
 	return nil
 }
 
+// HTTPRequester allows a net/http.Client to be mocked for tests.
+type HTTPRequester interface {
```

Comment:
1. Can we avoid exporting this? `lib/auth` should not present an interface for generic HTTP requests to the rest of Teleport
2. Better yet, can we avoid making an interface here at all? There are a number of ways you can test an HTTP call without this (spin up an http test server, provide a custom roundtripper, etc)

Reply:
For now I'll just unexport this interface, that's a good call. I'll add a comment to note that this should be changed in the future. I've been using Go for awhile now but somehow almost never used the `net/http` package, so testing it properly is new to me, thanks for the tips.